### ISSUE TITLE
Example for highlighting & text selection?

### ISSUE THREAD

**Issue by asadath1395** [wojtekmaj/react-pdf#279]
Hi

Can you give an example on how to select and highlight from the PDF?

**Comment by wojtekmaj** [comment]
Regarding highlight:
#212, #235 

Regarding selection:
Text rendered by text layer is just plain HTML, so you can look for generic solutions online. However, as far as I know it's not possible to programatically select text on the page. You can, though, get selection after the user selected the text if you use `window.getSelection()`.

**Comment by wojtekmaj** [comment]
I'm closing this issue to give other issues more visibility. If you still need assistance with this matter, please do not hesitate to reply and I'll reopen.

Happy coding!

**Comment by Samrat-Saha-Sammy** [comment]
@wojtekmaj  could you please explain this line? 
`makeTextRenderer = searchText => textItem => highlightPattern(textItem.str, searchText);`
How the textItem is been retreived?

**Comment by wojtekmaj** [comment]
This is a function that returns a function that's gonna be passed to React-PDF. Because of that, every time search text is changed you get a new function and React knows it should update the component that received it.

**Comment by Samrat-Saha-Sammy** [comment]
So, 
we used this code and we are getting an error as below. 
`makeTextRenderer = searchText => textItem => highlightPattern(textItem.str, searchText);`


ERROR: 
`in ./src/components/PdfViewer/PdfViewer.js
Module build failed: SyntaxError: C:/Users/santosh.gautam/Desktop/santosh/rsa/riskadjusmentui/src/components/PdfViewer/PdfViewer.js: Unexpected token (47:19)
  45 |   //   }
  46 |   // } > 47 |   makeTextRenderer = searchText => textItem => highlightPattern(textItem.str, searchText);
     |                    ^
  48 | //   makeTextRenderer  (searchText) {
  49 | //   return (textItem) => {
  50 | //     debugger;
`

**then we restructured the code as below:** 
`makeTextRenderer  (searchText) {
  return (textItem) => {
    highlightPattern(textItem.str, searchText);
  }
}`
Now it is not working. No error no highlight text

**Comment by vvvprabhakar** [comment]
+1

**Comment by wojtekmaj** [comment]
Looks like you have an odd environment which doesn't recognize arrow functions. You did well, but you changed one if two arrow functions into normal ones.

**Comment by vvvprabhakar** [comment]
could you please suggest a code other than this ?

**Comment by wojtekmaj** [comment]
Told you you did well, but left it halfway done :D Just change the other function too. Something like this could work.

```js
makeTextRenderer  (searchText) {
  return function (textItem) {
    return highlightPattern(textItem.str, searchText);
  }
}
```

**Comment by vvvprabhakar** [comment]
Its not working  !!

**Comment by wojtekmaj** [comment]
I'm sorry that examples I gave are not working for you. I re-checked and used my [test suite](https://github.com/wojtekmaj/react-pdf/tree/master/test) as a base, and replaced `Test.jsx` content with the recipe given above. It definitely does work.

It must be some problem with either your app configuration (example I gave assumes `@babel/plugin-proposal-class-properties` being used - I should fix it actually) or with your browser. If that's not it, you're on your own, that's way too few information to figure anything else.
